src/PayID/pay-id-client.ts

```diff
@@ -30,28 +44,54 @@ export default class PayIDClient {
     // Swagger generates the '/' in the URL by default and the payment pointer's 'path' is prefixed by a '/'. Strip off the leading '/'.
     const path = paymentPointer.path.substring(1)
 
-    // TODO(keefertaylor): Append headers here.
     const client = new ApiClient()
     client.basePath = `https://${paymentPointer.host}`
-    const apiInstance = new DefaultApi(client)
+
+    // Accept only the given netowrk in response.
+    const accepts = [`application/xrpl-${network}+json`]
 
     return new Promise((resolve, reject) => {
-      // Ignore unused var in generated code.
-      apiInstance.resolvePayID(path, (error, data, _response) => {
-        if (error) {
-          if (error.status === 404) {
-            // Not Found
-            resolve(undefined)
+      // NOTE: Swagger produces a higher level client that does not require this level of configuration,
+      // however access to Accept headers is not available unless we access the underlying class.
+      const postBody = null
+      const pathParams = {
+        path,
+      }
+      const queryParams = {}
+      const headerParams = {}
+      const formParams = {}
+      const authNames = []
+      const contentTypes = []
+      const returnType = Mapping
+      client.callApi(
+        '/{path}',
+        'GET',
+        pathParams,
+        queryParams,
+        headerParams,
+        formParams,
+        postBody,
+        authNames,
+        contentTypes,
+        accepts,
+        returnType,
+        (error, data, _response) => {
+          if (error) {
+            if (error.status === 404) {
+              // Not Found
+              resolve(undefined)
+            } else {
+              const message = `${error.status}: ${error.response.text}`
+              reject(new PayIDError(PayIDErrorType.UnexpectedResponse, message))
```

Comment:
Whoa. That's a lot of properties!

Reply:
Assume you're talking about the `callApi` method? 

Yeah. The parsing in Swagger wants to use `accept` headers as an array of content types the client may accept (and the server picks one and returns it). This isn't really analogous to the PayID protocol, where we're asking for a specific thing (an address of network / asset) rather than an array of things.

Swagger wants us to declare content types up front and then ranks them in their declared order.  I'm not really sure if Swagger is abusing the semantics of the accept header or if PayID is, but

Reply:
> Yeah. The parsing in Swagger wants to use accept headers as an array of content types the client may accept

That is valid. As Doug pointed out (and I learned today), it's fine to pass an Accept header for `btc+json` and respond with `Content-Type = xrpl+json`. The web is just like that. (Although I don't think that's a great implementation).

It's also reasonable to pass an array of `Accept` headers that you would accept, in order of preference. For example, maybe you have XRP, ETH, and BTC wallets, and you would want to pay someone in that order based on transaction costs. They don't have a XRP wallet, but they have an ETH address, so they respond with ETH.

This is better than responding with `eth+json` when you ONLY asked for `xrpl+json`, because at least `ETH` was in the array of Accept headers you passed.